sympy/vector/lamecoeff.py

```diff
@@ -0,0 +1,77 @@
+from sympy.core.basic import Basic
+from sympy import sqrt, sin, cos
+from sympy.core.function import Derivative
+
+
+class CoeffProvider(Basic):
+    """
+    It provides specific information of curvilinear coordinate
+    system to CoordSysCartesian.
+
+    Ideally, users should not instantiate this class.
+
+    """
+
+    def __new__(cls, system, curv_coord_name=None, transformation_equations=None):
+        obj = super(CoeffProvider, cls).__new__(cls, system)
+        obj.sys = system
+        obj._x, obj._y, obj._z = system.x, system.y, system.z
+        if curv_coord_name is not None:
+            class DefinedCoefficient:
```

Comment:
Too complicated code, can be rewritten in a simpler form. What about just use some static private dictionaries in the class namespace? Like
```python
class CoeffProvider(...):
    _coefficient_mapping = dict( ... )
    _curvilinear_parameters = dict( ... )

    def __new__( ... ):
        ...
```